packages/nextra/src/server/mdx-plugins/rehype-better-react-mathjax.ts

```diff
@@ -0,0 +1,168 @@
+import type { MdxJsxAttribute } from 'hast-util-to-estree/lib'
+import { toText } from 'hast-util-to-text'
+import type { Plugin } from 'unified'
+import { SKIP, visitParents } from 'unist-util-visit-parents'
+
+export type Options = { src?: string; config?: MathJax3Config } | undefined
+
+const MATHJAX_IMPORTS = {
+  type: 'mdxjsEsm',
+  data: {
+    estree: {
+      body: [
+        {
+          type: 'ImportDeclaration',
+          source: { type: 'Literal', value: 'nextra/components' },
+          specifiers: ['MathJax', 'MathJaxContext'].map(name => ({
+            type: 'ImportSpecifier',
+            imported: { type: 'Identifier', name },
+            local: { type: 'Identifier', name }
+          }))
+        } satisfies ImportDeclaration
+      ]
+    }
+  }
+}
+
+function wrapInMathJaxContext(
+  children: RootContent[],
+  { config, src }: NonNullable<Options>
+) {
+  const attributes: MdxJsxAttribute[] = []
+  if (src) {
+    attributes.push({ type: 'mdxJsxAttribute', name: 'src', value: src })
+  }
+  if (config && Object.keys(config).length) {
+    attributes.push({
+      type: 'mdxJsxAttribute',
+      name: 'config',
+      value: {
+        type: 'mdxJsxAttributeValueExpression',
+        value: '',
+        data: {
+          estree: {
+            type: 'Program',
+            sourceType: 'module',
+            body: [
+              {
+                type: 'ExpressionStatement',
+                expression: valueToEstree(config)
+              }
+            ]
+          }
+        }
+      }
+    })
+  }
+
+  return {
+    type: 'mdxJsxFlowElement',
+    name: 'MathJaxContext',
+    attributes,
+    children
+  }
+}
+
+/**
+ * Wrap the math in the appropriate braces. Defaults to `\(...\)` for inline and `\[...\]` for display,
+ * but will use the braces provided by `options` if they are present.
+ */
+function wrapInBraces(
+  source: string,
+  displayMath: boolean,
+  options: NonNullable<Options>
+): string {
+  const inlineBraces = options?.config?.tex?.inlineMath?.[0] || ['\\(', '\\)']
+  const displayBraces = options?.config?.tex?.displayMath?.[0] || ['\\[', '\\]']
+  const [before, after] = displayMath ? displayBraces : inlineBraces
+  return `${before}${source}${after}`
+}
+
+/**
+ * Wraps math in a `<MathJax>` component so that it can be rendered by `better-react-mathjax`.
+ */
+export const rehypeBetterReactMathjax: Plugin<
+  [Opts: Options, isRemoteContent: boolean],
+  Root
+> =
+  (options = {}, isRemoteContent) =>
+  tree => {
+    let hasMathJax = false
+
+    visitParents(tree, 'element', (element, parents) => {
+      const classes = Array.isArray(element.properties.className)
+        ? element.properties.className
+        : []
+      // This class can be generated from markdown with ` ```math `.
+      const languageMath = classes.includes('language-math')
+      // This class is used by `remark-math` for flow math (block, `$$\nmath\n$$`).
+      const mathDisplay = classes.includes('math-display')
```

Comment:
I will remove this, because tests fails

<img width="426" alt="image" src="https://github.com/shuding/nextra/assets/7361780/a6d01d50-96ad-4be9-bfc6-8ce4822ad0a5">
<img width="468" alt="image" src="https://github.com/shuding/nextra/assets/7361780/67d7fdc4-3fca-4031-bfa1-324af578b4f6">